pkg/encryption/cipher_test.go

```diff
@@ -51,12 +53,152 @@ func TestEncodeAndDecodeAccessTokenB64(t *testing.T) {
 	c, err := NewCipher([]byte(secret))
 	assert.Equal(t, nil, err)
 
-	encoded, err := c.Encrypt(token)
+	encoded, err := c.Encrypt([]byte(token))
 	assert.Equal(t, nil, err)
 
 	decoded, err := c.Decrypt(encoded)
 	assert.Equal(t, nil, err)
 
-	assert.NotEqual(t, token, encoded)
-	assert.Equal(t, token, decoded)
+	assert.NotEqual(t, []byte(token), encoded)
+	assert.Equal(t, []byte(token), decoded)
+}
+
+func TestEncryptAndDecrypt(t *testing.T) {
+	var err error
+
+	// Test our 3 cipher types
```

Comment:
Could you expand on this, took me a moment to realise that `NewCipher` does the Base64 testing, perhaps we could be more explicit by wrapping both types of cipher in base64 for testing as well?

```
base64WrappedCipher := func(newCipher func([]byte) (Cipher, error)) func([]byte) (Cipher, error) {
  return func(b []byte) (Cipher, error) {
    c, err := newCipher(b)
     if err != nil {
      return nil, err
     }
     return NewBase64Cipher(c)
  }
}
```

Reply:
I like that design. I took that motivation and made the newBase64Cipher operate on Cipher init methods itself here:

https://github.com/oauth2-proxy/oauth2-proxy/pull/539/commits/fc418bfd37a7728a936cd61df512d8d3f49142de